### ISSUE TITLE
Can not do matrix multiply

### ISSUE THREAD

**Issue by elite-sheep** [twistedfall/opencv-rust#107]
OS: Ubuntu 16.04
OpenCV Version: 4.2

I was trying to do matrix multiplication in my project, and in that file I included the following things:  

```
use opencv::core::{Mat, MatExpr, MatExprTrait, Matx_MulOp, Vec3b, Vec3f};
use opencv::prelude::*;
use opencv::types::VectorOfMat;
use std::error::Error;
```

And when I tried to do the multiplication, I got the following answer:  

```
error[E0369]: binary operation `*` cannot be applied to type `opencv::opencv::hub::core::MatExpr`
   --> src/./core/debevec_crf_solver.rs:104:54
    |
104 |     let X: Mat = (MatExpr::from_mat(&A_inv).unwrap() * MatExpr::from_mat(&B).unwrap()).to_mat().unwrap();
    |                   ---------------------------------- ^ ------------------------------ opencv::opencv::hub::core::MatExpr
    |                   |
    |                   opencv::opencv::hub::core::MatExpr
    |
    = note: an implementation of `std::ops::Mul` might be missing for `opencv::opencv::hub::core::MatExpr`

error: aborting due to previous error
```

I looked into the code and it seems that we do not implement such ops, or am I doing something wrong here?

**Comment by twistedfall** [comment]
For your case you should be able to use [Mat.mul](https://docs.rs/opencv/0.33.0/opencv/core/prelude/trait.MatTrait.html#method.mul). But I'll definitely look into supporting more C++ operators like multiplication so please don't close this issue.

**Comment by elite-sheep** [comment]
@twistedfall Hi, from my point of view, Mat.mul is not what I am looking for. It is a element multiply. Like A(M*N).mul(B(M*N)) => C(M*N) but I am looking for matrix multiplication, A(M*N)*B(N*P) = C(M*P)
At last I decided to write such a function in my project. And I am willing to contribute if you have any plan regarding this.

**Comment by twistedfall** [comment]
Ah, I see, my bad. Are you saying that you implemented it manually on base of `Mat`? If so it would be useful to have it as a code snippet, but I generally want to avoid adding functionality already present in OpenCV, it's better to expose it.

**Comment by elite-sheep** [comment]
Yes, that's just what I expected. By the way, I opened this issue because I saw the doc of [MatExprTrait](https://docs.rs/opencv/0.33.0/opencv/core/trait.MatExprTrait.html) claims that it implemented such function. Actually we did not. Thanks for your clarifying again.

**Comment by twistedfall** [comment]
Yeah, the docs are copied directly from c++ headers so they contain references to possibly not implemented functionality and for example c++ code samples.

**Comment by twistedfall** [comment]
Implemented in v0.36. MatExpr by MatExpr multiplication is done by `opencv::core::mul_matexpr_matexpr` function.
